plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSource.java

```diff
@@ -359,6 +364,45 @@ public Block block(Page sourcePage, MaskDeletedRowsFunction maskDeletedRowsFunct
         }
     }
 
+    /**
+     * This ColumnAdaptation creates a RowBlock column containing the three
+     * ACID columms derived from the startingRowId and bucketId, and a special
+     * original files transaction block, plus a block containing
+     * all the columns not changed by the UPDATE statement.
+     */
+    private static final class UpdatedRowAdaptationWithOriginalFiles
+            implements ColumnAdaptation
+    {
+        private final long startingRowId;
+        private final Block bucketBlock;
+        private final HiveUpdateProcessor updateProcessor;
+        private final List<Integer> nonUpdatedSourceChannels;
+
+        public UpdatedRowAdaptationWithOriginalFiles(long startingRowId, int bucketId, HiveUpdateProcessor updateProcessor, List<HiveColumnHandle> dependencyColumns)
+        {
+            this.startingRowId = startingRowId;
+            this.bucketBlock = nativeValueToBlock(INTEGER, Long.valueOf(computeBucketValue(bucketId, 0)));
+            this.updateProcessor = updateProcessor;
+            this.nonUpdatedSourceChannels = updateProcessor.makeNonUpdatedSourceChannels(dependencyColumns);
```

Comment:
nit: use `requireNotNull` on `updateProcessor` and `dependencyColumns`

Reply:
Good idea; done.